// @amalto/help/build/index.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var classNames = require("classnames");
var Help = (function (_super) {
    __extends(Help, _super);
    function Help() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this._helpPopup = HTMLSpanElement = null;
        return _this;
    }
    Help.prototype.render = function () {
        var _this = this;
        return (React.createElement("span", { className: classNames('fas fa-fw fa-question-circle default-color', this.props.containerClass), "data-content": this.props.text, ref: function (dom) { return _this._helpPopup = dom; }, style: this.props.style }));
    };
    Help.prototype.componentDidMount = function () {
        $(this._helpPopup).popover({
            container: 'body',
            trigger: 'hover',
            html: true
        });
    };
    Help.prototype.componentWillUnmount = function () {
        $(this._helpPopup).popover('destroy');
    };
    return Help;
}(React.Component));
exports.default = Help;
